kornia/augmentation/functional.py

```diff
@@ -525,3 +526,139 @@ def _apply_crop(input: torch.Tensor, params: Dict[str, torch.Tensor], return_tra
+    and size by input.size()
+
+    Args:
+        input (torch.Tensor): input image.
+        params Dict[str, torch.Tensor]:
+            params['widths'] must be widths tensor
+            params['heights'] must be heights tensor
+            params['xs'] must be x positions tensor
+            params['ys'] must be y positions tensor
+            params['values'] is the value to fill in
+    """
+    if not (params['widths'].size() == params['heights'].size() == params['xs'].size() == params['ys'].size()):
+        raise TypeError(
+            f"''rectangle params components must have same shape"
+        )
+
+    mask = torch.zeros(input.size()).type_as(input)
+    values = torch.zeros(input.size()).type_as(input)
+
+    widths = params['widths']
+    heights = params['heights']
+    xs = params['xs']
+    ys = params['ys']
+    vs = params['values']
+    for i_elem in range(input.size()[0]):
+        h = widths[i_elem].item()
+        w = heights[i_elem].item()
+        y = ys[i_elem].item()
+        x = xs[i_elem].item()
+        v = vs[i_elem].item()
+        mask[i_elem, :, int(y):int(y + w), int(x):int(x + h)] = 1.
+        values[i_elem, :, int(y):int(y + w), int(x):int(x + h)] = v
+    if return_transform:
+        raise NotImplementedError
+    return torch.where(mask == 1., values, input)
+
+
+def apply_adjust_brightness(input: torch.Tensor, params: Dict[str, torch.Tensor],
+                            return_transform: bool = False):
+    """ Wrapper for adjust_brightness for Torchvision-like param settings.
+
+    Args:
+        input (torch.Tensor): Image/Input to be adjusted in the shape of (*, N).
+        brightness_factor (Union[float, torch.Tensor]): Brightness adjust factor per element
+          in the batch. 0 gives a black image, 1 does not modify the input image and 2 gives a
+          white image, while any other number modify the brightness.
+
+    Returns:
+        torch.Tensor: Adjusted image.
+    """
+    if return_transform:
+        raise NotImplementedError
+    return adjust_brightness(input, params['brightness_factor'].type_as(input) - 1)
+
+
+def apply_adjust_contrast(input: torch.Tensor, params: Dict[str, torch.Tensor],
+                          return_transform: bool = False):
+    """Wrapper for adjust_contrast for Torchvision-like param settings.
+    Args:
+        input (torch.Tensor): Image to be adjusted in the shape of (*, N).
+        params['contrast_factor'] (Union[float, torch.Tensor]):
+          Contrast adjust factor per element in the batch.
+          0 generates a compleatly black image, 1 does not modify
+          the input image while any other non-negative number modify the
+          brightness by this factor.
+
+    Returns:
+        torch.Tensor: Adjusted image.
+    """
+    if return_transform:
+        raise NotImplementedError
+    return adjust_contrast(input, params['contrast_factor'].type_as(input))
```

Comment:
just types_as, what about device ?

`return adjust_contrast(input, params['contrast_factor'].to(input.device, input.dtype))`

Reply:
Actually, I found type_as will copy both input.dtype and input.device. You may try it out. If it worked, I suggest we should move from the ```.to(input.type).to(input.device)``` to ```type_as(input)```. It is more concise.

Reply:
Amazing found 😍